crates/evm/src/ef_tests/cases/blockchain_test.rs

```diff
@@ -0,0 +1,282 @@
+                        | ForkSpec::MergeEOF
+                        | ForkSpec::MergeMeterInitCode
+                        | ForkSpec::MergePush0
+                        | ForkSpec::Unknown
+                )
+            })
+            .par_bridge()
+            .try_for_each(|case| {
+                // Create a new test database and initialize a provider for the test case.
+                let db = create_test_rw_db();
+                let (_static_files_dir, static_files_dir_path) = create_test_static_files_dir();
+                let provider = ProviderFactory::new(
+                    db.as_ref(),
+                    Arc::new(case.network.clone().into()),
+                    static_files_dir_path,
+                )?
+                .provider_rw()
+                .unwrap();
+
+                // Insert initial test state into the provider.
+                provider
+                    .insert_historical_block(
+                        SealedBlock::new(
+                            case.genesis_block_header.clone().into(),
+                            BlockBody::default(),
+                        )
+                        .try_seal_with_senders()
+                        .unwrap(),
+                        None,
+                    )
+                    .map_err(|err| Error::RethError(err.into()))?;
+                case.pre.write_to_db(provider.tx_ref())?;
+
+                // Initialize receipts static file with genesis
+                {
+                    let mut receipts_writer = provider
+                        .static_file_provider()
+                        .latest_writer(StaticFileSegment::Receipts)
+                        .unwrap();
+                    receipts_writer
+                        .increment_block(StaticFileSegment::Receipts, 0)
+                        .unwrap();
+                    receipts_writer.commit_without_sync_all().unwrap();
+                }
+
+                // Decode and insert blocks, creating a chain of blocks for the test case.
+                let last_block = case.blocks.iter().try_fold(None, |_, block| {
+                    let decoded = SealedBlock::decode(&mut block.rlp.as_ref())?;
+                    provider
+                        .insert_historical_block(
+                            decoded.clone().try_seal_with_senders().unwrap(),
+                            None,
+                        )
+                        .map_err(|err| Error::RethError(err.into()))?;
+                    Ok::<Option<SealedBlock>, Error>(Some(decoded))
+                })?;
+                provider
+                    .static_file_provider()
+                    .latest_writer(StaticFileSegment::Headers)
+                    .unwrap()
+                    .commit_without_sync_all()
+                    .unwrap();
+
+                let (evm, working_set, storage) = get_evm_with_storage(&EvmConfig::default());
+                let root = commit(working_set, storage.clone());
```

Comment:
call finalize hook here after commit

Reply:
`commit` is not even needed here as it's already done in `get_evm_with_storage`... i just did it to get the genesis root BUT i was able to figure out i can use `GENESIS_STATE_ROOT` for that.